TFUtil.py

```diff
@@ -1249,6 +1249,54 @@ def copy_template(self, name=None, dtype=None):
       kwargs["dtype"] = dtype
     return Data(**kwargs)
 
+  def copy_template_excluding_axis(self, exclude_axis, name=None):
+    """
+    :param str|None name: if set, this will be the new name.
+    :param int exclude_axis: axis to be removed.
+    :return: copy of myself excluding remove_axis_num axis, without placeholder.
+    :rtype: Data
+    """
+    kwargs = self.get_kwargs()
+    if exclude_axis < 0:
+      exclude_axis += self.batch_ndim
+      assert exclude_axis >= 0
+    assert 0 <= exclude_axis < self.batch_ndim
+    exclude_batch_dim_axis = exclude_axis == self.batch_dim_axis
+    if not exclude_batch_dim_axis:
+      axis_to_exclude_wo_b = self.get_batch_axis_excluding_batch(exclude_axis)
```

Comment:
Never assign a variable only conditionally (in `if` but not in `else`).
Anyway, you don't need the `if` at all. Just remove the `if`.